src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -167,45 +127,63 @@
+    @ApiOperation(value = "Retrieves all or specific subscriptions")
+    public ResponseEntity<?> getSubscriptions(@RequestParam(required = false) String subscriptionName) {
+        LOGGER.debug("Fetching subscriptions has been initiated");
+        if(subscriptionName == null || subscriptionName.isEmpty()) {
+            return getAllSubscriptions();
+        } else {
+            return getListedSubscriptions(subscriptionName);
+        }
+    }
 
-        subscriptionNamesList.forEach(subscriptionName -> {
-            LOGGER.debug("Subscription deleting has been started: {}", subscriptionName);
+    @Override
+    @CrossOrigin
+    @ApiOperation(value = "Retrieve a subscription")
+    public ResponseEntity<?> getSubscriptionByName(@PathVariable String subscriptionName) {
+        return getSingleSubscription(subscriptionName);
+    }
 
-            try {
-                if (subscriptionService.deleteSubscription(subscriptionName)) {
-                    LOGGER.debug("Subscription was deleted successfully: {}", subscriptionName);
-                } else {
-                    LOGGER.error("Subscription to delete was not found: {}", subscriptionName);
-                    errorMap.put(subscriptionName, SUBSCRIPTION_NOT_FOUND);
-                }
-            } catch (AccessException e) {
-                LOGGER.error("Failed to delete subscription: {}", subscriptionName, e);
-                errorMap.put(subscriptionName, INVALID_USER);
-            } catch (Exception e) {
-                LOGGER.error("Failed to delete subscriptions.", e);
-                errorMap.put(subscriptionName, e.getClass().toString() + " : " + e.getMessage());
-            }
-        });
-        return getDeleteResponse(errorMap);
+    @Override
+    @CrossOrigin
+    @ApiOperation(value = "Remove subscription(s)")
+    public ResponseEntity<?> deleteSubscriptions(@RequestParam String subscriptionName) {
+        return deleteListedSubscriptions(subscriptionName);
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Retrieves all subscriptions")
-    public ResponseEntity<?> getSubscriptions() {
-        LOGGER.debug("Fetching subscriptions has been initiated");
+    @ApiOperation(value = "Remove a subscription")
+    public ResponseEntity<?> deleteSubscriptionByName(@PathVariable String subscriptionName) {
+        return deleteSingleSubscription(subscriptionName);
+    }
+
+    private ResponseEntity<?> getSingleSubscription(String subscriptionName) {
+        try {
+            Subscription subscription = subscriptionService.getSubscription(subscriptionName);
+            subscription.setPassword("");
+            return new ResponseEntity<>(subscription, HttpStatus.OK);
+        } catch (SubscriptionNotFoundException e) {
+            String errorMessage = "Subscription not found: " + subscriptionName;
+            LOGGER.error(errorMessage);
+            LOGGER.debug("Subscription not found traceback:\n {}", e);
```

Comment:
If we separate the logging, at least include the subscription name in the debug message - since it can be separated in the logs and impossible to find otherwise

Reply:
Haha, yes it was written like that before and I thought it was odd looking but decided to just leave it. I'm just remove the debug line altogether.